src/devices/LedDisplay/Large4Digit7SegmentDisplay.cs

```diff
@@ -0,0 +1,310 @@
+            2 => (int)Address.Digit3,
+            3 => (int)Address.Digit4,
+            _ => throw new ArgumentOutOfRangeException(nameof(digit)),
+        };
+        #endregion
+
+        #region Public members
+        #region Enums
+        /// <summary>
+        /// Dot display setup
+        /// </summary>
+        /// <remarks>Specific to device</remarks>
+        [Flags]
+        public enum Dot : byte
+        {
+            /// <summary>
+            /// Disable all dots
+            /// </summary>
+            Off = 0b0000_0000,
+            /// <summary>
+            /// Center colon
+            /// </summary>
+            CenterColon = 0b0000_0010,
+            /// <summary>
+            /// Left colon
+            /// </summary>
+            LeftColon = LeftLower | LeftUpper,
+            /// <summary>
+            /// Left lower dot
+            /// </summary>
+            LeftLower = 0b0000_0100,
+            /// <summary>
+            /// Left upper dot
+            /// </summary>
+            LeftUpper = 0b0000_1000,
+            /// <summary>
+            /// Decimal point (between third and fourth digits)
+            /// </summary>
+            DecimalPoint = 0b0001_0000
+        }
+        #endregion
+
+        /// <summary>
+        /// Initialize 7-Segment display
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Large4Digit7SegmentDisplay(I2cDevice i2cDevice) : base(i2cDevice) { }
+
+        /// <inheritdoc/>
+        public int NumberOfDigits { get; } = MaxNumberOfDigits;
+
+        /// <summary>
+        /// Gets or sets a single digit's segments by id
+        /// </summary>
+        /// <param name="address">digit address (0..3)</param>
+        /// <returns>Segment in display buffer for the given address</returns>
+        public Segment this[int address]
+        {
+            get => (Segment)_displayBuffer[TranslateDigitToBufferAddress(address)];
+            set => _displayBuffer[TranslateDigitToBufferAddress(address)] = (byte)value;
+        }
+
+        /// <remarks>Write clears dots, you'll have to reset them after you've </remarks>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="startAddress"/></exception>
+        /// <inheritdoc/>
+        public override void Write(ReadOnlySpan<byte> digits, int startAddress = 0)
+        {
+            if (digits.Length == 0)
+            {
+                // Nothing to write
+                return;
+            }
+
+            if (startAddress < 0 || startAddress >= MaxNumberOfDigits)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startAddress));
+            }
+
+            if (digits.Length + startAddress > MaxNumberOfDigits)
+            {
+                throw new NotSupportedException($"{nameof(Large4Digit7SegmentDisplay)} only supports {MaxNumberOfDigits - startAddress} digits starting from address {startAddress}");
```

Comment:
should this be `ArgumentOutOfRangeException` too? If so this and previous ifs can also be joined together